rust/src/detect_parser/error.rs

```diff
@@ -0,0 +1,37 @@
+/* Copyright (C) 2022 Open Information Security Foundation
```

Comment:
I would have chosen `detect` instead of `detect_parser` and moved detect.rs into it (in its own commit)

Reply:
Good idea.